### ISSUE TITLE
Visualizing dataframe with non-EPSG:4326 CRS is failing silently

### ISSUE THREAD

**Issue by jorisvandenbossche** [CartoDB/cartoframes#1274]
When trying to map a GeoDataFrame with projected coordinates (not lat/long EPSG:4326), such as this example:

```
import geopandas
nybb = geopandas.read_file(geopandas.datasets.get_path("nybb"))

from cartoframes.viz import Map, Layer
Map(Layer(nybb))
```

you get a plot area, but with just a grey background (which is probably somewhere in the sea). 

Since GeoDataFrames have a `crs` attribute, I think it would be helpful for users to check this, and either raise an informative error message, or either automatically transform the data when visualizing.

**Comment by Jesus89** [comment]
Yes, this is a good point, because actually we are not using the CRS either for the visualization or the upload to our platform. It assumes that everything is 4326.

We are also extracting the SRID in the [decoding](https://github.com/CartoDB/cartoframes/blob/develop/cartoframes/utils/geom_utils.py#L28) but it's also not used.

So this is a very good feature we need to integrate. Could you provide some guidance or a PoC about how to integrate and use spatial reference systems in the library? Thanks!

**Comment by jorisvandenbossche** [comment]
> Could you provide some guidance or a PoC about how to integrate and use spatial reference systems in the library? Thanks!

I don't know the online Carto application very well, but I suppose there are good reasons to keep the data on the server always in the same (4326) CRS? 

For the local usage of Cartoframes, I think there are basically two options:

- Raise an informative error if the CRS is not EPSG:4326 and let the user convert themselves
- Automatically convert the geometries to EPSG:4326 if crs information is available

For both cases, you first need to check if the CRS is equal to CRS:4326. This can be done with using `pyproj`, but is actually not fully straightforward right now because in geopandas we are not yet using pyproj.CRS as the `crs` representation (but which is coming: https://github.com/geopandas/geopandas/issues/1003, https://github.com/geopandas/geopandas/pull/1101). This does the trick more or less:

```
import pyproj
pyproj.CRS.from_user_input(gdf.crs) == "EPSG:4326"
```

but because many people are probably using the (soon to be deprecated) "+init" method (where the `GeoDataFrame.crs` is set to eg `{'init': 'epsg':4326'}`), the above check will fail in this specific case. So you might want to also compare the GeoDataFrame's crs to `pyproj.CRS.from_user_input({'init': 'epsg':4326'})` in addition (the difference between both is the "axis order", which is something that is handled by geopandas when converting to another CRS, but for checking the equality of two CRS, it gives a difference).

If you would like to automatically convert the geometries to EPSG:4326 (so the user does not need to do this themselves), you can use the `to_crs` functionality of GeoPandas:

```
gdf_wgs84 = gdf.to_crs(epsg=4326)
```

**Comment by Jesus89** [comment]
https://app.clubhouse.io/cartoteam/story/93706/visualize-non-4326-datasets
